test/e2e/masteruserrecord_test.go

```diff
@@ -73,6 +73,36 @@ func TestMasterUserRecord(t *testing.T) {
 		// TODO: verify expected condition when the member operator has a logic that updates NsTemplateSet and its status
 		verifyResources(awaitility, extraMur)
 	})
+
+	t.Run("delete MasterUserRecord and expect UserAccount to be deleted", func(t *testing.T) {
+		// given
+		currentMur := NewHostAwaitility(awaitility).GetMasterUserRecord(mur.Name)
```

Comment:
1. Wait for mur & user accounts to be created first.
2. Then delete and wait for all the resources to be deleted.

You already have 2 but missed 1. If you don't do 1 then 2 doesn't guarantee that you user accounts were deleted (maybe they controller hasn't created them yet).

Reply:
I would say that 1. is part of the test "setup" at the beginning of the whole test: 
https://github.com/codeready-toolchain/host-operator/pull/50/files/564e8c7285b0150c7aef2714e5258ef582104585..e50c92515c2d042e29529eed3df2fbcccd50d060#diff-20c771505f30e89a2a3e8efdee8c003fR29-R36
notice that it's not a subtest, in other words, it will be executed every time even if you run only this subtest

Reply:
Oh... right. I overlooked that. Never mind then :)